parser.c

```diff
@@ -1015,3 +1015,31 @@ void validate_access_oTfd(int timeidx, int pos, int line, const char * function,
 		logg("             found in %s() (line %i) in %s", function, line, file);
 	}
 }
+
+void reresolveHostnames(void)
+{
+	int clientID;
+	for(clientID = 0; clientID < counters.clients_MAX; clientID++)
+	{
+		// Process this client only if it has at least one active query in the log
+		if(clients[clientID].count < 1)
+			continue;
+
+		// Get client hostname
+		char *hostname = resolveHostname(clients[clientID].ip);
+		if(strlen(hostname) > 0)
+		{
+			// Delete possibly already existing hostname pointer before storing new data
+			if(clients[clientID].name != NULL)
+			{
+				free(clients[clientID].name);
+				clients[clientID].name = NULL;
+			}
+
+			// Store client hostname
+			clients[clientID].name = strdup(hostname);
+			memory.clientnames += (strlen(hostname) + 1) * sizeof(char);
```

Comment:
Shouldn't you also decrement the memory usage when you free the old hostname?

Reply:
Yes, I'll add that.

Reply:
Added in 6445020